### ISSUE TITLE
[Bug] An element with the same key already exists in Namescope

### ISSUE THREAD

**Issue by yrest** [xamarin/Xamarin.Forms#13767]
Original problem reported more than a year ago - https://github.com/xamarin/Xamarin.Forms/issues/9366
Suggested NameScope.SetNameScope(bindable, null); has no effect.

###Description
From Play Store on released version, running Xamarin.Forms.Xaml.Extensions.LoadFromXaml causes subject exception. Development / debugging works as expected.

###Steps to Reproduce
Release app
Call Xamarin.Forms.Xaml.Extensions.LoadFromXaml to load custom page
Observe failure

###Expected Behavior
Page is loaded correctly

###Actual Behavior
Generates exception

###Basic Information
Version with issue: 5.0.0.2012
Last known good version: 4.1.0.555618
IDE: VS 2019
Platform Target Frameworks:
Android: 11.0

###Workaround
There's no workaround anymore, because targeting older Android is not supported by Play Store.

**Comment by hartez** [comment]
@yrest Can you provide a minimal repro project which demonstrates this issue? Or at least provide the XAML you are loading? 

> Development / debugging works as expected.

Do the versions released in the Play Store call LoadFromXaml using the same XAML as the debug version?

**Comment by yrest** [comment]
@hartez thanks for your attention. I accept, I could be doing it all wrong, but it was working up until 4.2 and just might have been fixed being itself a bug. The idea behind this scenario is to support multiple UIs - one standard for general public and then for dedicated users to have a version of xaml customized to their needs, which will be loaded when required. Unfortunately it only stops working in released app and it would be difficult to provide sample project. What we did for now is embedded all UI variants into the app and switching between them when needed, so immediate issue is bypassed. 

It was working as following:

// create new "default" page
Pages.Tablet.ItemListPage itemList = new Pages.Tablet.ItemListPage();
// check if custom xaml is loaded
if (!String.IsNullOrWhiteSpace(targetXaml))
        // reload new xaml
        itemList = Xamarin.Forms.Xaml.Extensions.LoadFromXaml(itemList, targetXaml);
// launch the target page
await App.MainNavigation.PushAsync(itemList);

In code behind page constructor i'm also initializing some other elements
What happens now is the LoadFromXaml generates exception (I think on ListView) that at 13:24 element already exist.

I was hoping someone who knows how the function is intended to work will say what had changes or perhaps this is simply not supported (anymore). Adding Xamarin.Forms.Internals.NameScope.SetNameScope(itemList, null); prior to loading xaml, doesnt help.

**Comment by yrest** [comment]
@hartez thanks for your attention. I accept, I could be doing it all wrong, but it was working up until 4.2 and just might have been fixed being itself a bug. The idea behind this scenario is to support multiple UIs - one standard for general public and then for dedicated users to have a version of xaml customized to their needs, which will be loaded when required. Unfortunately it only stops working in released app and it would be difficult to provide sample project. What we did for now is embedded all UI variants into the app and switching between them when needed, so immediate issue is bypassed. 

It was working as following:

// create new "default" page
Pages.Tablet.ItemListPage itemList = new Pages.Tablet.ItemListPage();
// check if custom xaml is loaded
if (!String.IsNullOrWhiteSpace(targetXaml))
        // reload new xaml
        itemList = Xamarin.Forms.Xaml.Extensions.LoadFromXaml(itemList, targetXaml);
// launch the target page
await App.MainNavigation.PushAsync(itemList);

In code behind page constructor i'm also initializing some other elements
What happens now is the LoadFromXaml generates exception (I think on ListView) that at 13:24 element already exist.

I was hoping someone who knows how the function is intended to work will say what had changes or perhaps this is simply not supported (anymore). Adding Xamarin.Forms.Internals.NameScope.SetNameScope(itemList, null); prior to loading xaml, doesnt help.
